addons/web/static/tests/helpers/mock_env.js

```diff
@@ -72,6 +72,11 @@ export function prepareRegistriesWithCleanup() {
     // better design for the way we clear these registries...
 }
 
+// This is exported in a utils object to allow for patching
+export const utils = {
+    prepareRegistriesWithCleanup,
```

Comment:
do you really need this? Do /web tests require your website registries to be set up to run? If it is only necessary for your tests, you can define another helper that you can call in the beforeEach hook of all your test suites.

Reply:
Hello, the main navbar is patched from the website module. In that patch we add the website service in its setup, so the website service needed to be mocked for all the tests
Here is the standalone commit: https://github.com/odoo/odoo/pull/89223/commits/8f6b092323cb5e1cc839e4dad4517823f2b39904

Reply:
I tried to patch only the function but could not find a way, I understood that patch can only be used on an object - but maybe I'm wrong - so used the utils object. Then the utils object imported in other files is the patched one, and when accessing  `const { prepareRegistriesWithCleanup } = utils;`, it will return the patched version of the function

Reply:
> Then the utils object imported in other files is the patched one, and when accessing const { prepareRegistriesWithCleanup } = utils;, it will return the patched version of the function

Ok then I don't understand how that magic is even possible but if you are sure and @aab-odoo does not deny it... I guess it's ok :)

Reply:
Indeed, you can't patch it without putting it in an object or an array. But even that way, it's not perfect, because if someone imports it and directly destructure it, for instance
```js
import { utils } from "...";
const { prepareRegistriesWithCleanup } = utils;
```
It doesn't work either.

But anyway, I was challenging the fact that you need to patch this function, instead of the way you're doing it. If you need to execute code before **your** tests, you can do that in before each (this is what we typically do to setup registries), and if you need to execute code before **all** tests (e.g. removing stuff from registries), you can directly do
```js
QUnit.testStart(() => {
   ...
});
```

Reply:
> But even that way, it's not perfect, because if someone imports it and directly destructure it, for instance [...]

That's what I have been worrying about in this discussion, is not it what is done here @younn-o ?

> If you need to execute code before your tests [...]

I guess @younn-o did that because /web tests were using the real website service breaking outside-of-website tests? Although, in what way was it breaking @younn-o ? Is not your patch of the navbar (using this website service) just supposed to say "I am not in the website app, do nothing" and thus not break anything ?

Reply:
We need to mock the website service for all the tests, because the `QUnit.testStart` from /web calls `prepareRegistriesWithCleanup` that will remove all the services from the service registry. 
Then the navbar (patched) in its setup will try to use the website service but won't find it.
@aab-odoo If we use `QUnit.testStart` from our website module and mock the website service from there, do we have a way to be sure that it will be executed after the /web `QUnit.testStart` ? Otherwise it could conflict with `prepareRegistriesWithCleanup` ?